components/public-api/gitpod/v1/prebuild.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+import "gitpod/v1/pagination.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service PrebuildService {
+    rpc StartPrebuild(StartPrebuildRequest) returns (StartPrebuildResponse) {}
+    rpc StopPrebuild(StopPrebuildRequest) returns (StopPrebuildResponse) {}
+
+    rpc GetPrebuild(GetPrebuildRequest) returns (GetPrebuildResponse) {}
+    rpc ListPrebuilds(ListPrebuildsRequest) returns (ListPrebuildsResponse) {}
+    rpc WatchPrebuild(WatchPrebuildRequest) returns (stream WatchPrebuildResponse) {}
+}
+
+message GetPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+message GetPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message ListPrebuildsRequest {
+    PaginationRequest pagination = 1;
+    string prebuild_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+
+message ListPrebuildsResponse {
+    PaginationResponse pagination = 1;
+    repeated Prebuild prebuilds = 2;
+}
+
+message StartPrebuildRequest {
+    string configuration_id = 1;
+    string branch = 2;
```

Comment:
This has the same ambiguity as https://github.com/gitpod-io/gitpod/pull/19062/files#r1390767255 I think.

Reply:
For starting a prebuild you need to provide the git reference (commit sha, tag, branch) you want to build.

Reply:
> For starting a prebuild you need to provide the git reference (commit sha, tag, branch) you want to build.

And which organization is that Prebuild belonging to? Are we expecting "configuration_id" to be passed as well?

Reply:
yes, we need configuration_id otherwise we don't know the repo

Reply:
>  yes, we need configuration_id otherwise we don't know the repo

Makes sense. Should this kind of information live in a comment in the proto? And we generate API docs from that? Or did we decide on sth else?

Reply:
Yes, communicating in the docs what is mandatory is my current understanding as well.

Reply:
so renaming it from `branch` to `ref` is enough?